clients/python/tests/test_client.py

```diff
@@ -1206,3 +1208,129 @@ async def test_as_mlops_engineer_i_would_like_to_store_a_longer_documentation_fo
     )
     assert da
     assert da.uri == "https://README.md"
+
+
+class MockPath:
+    def __init__(self, path_arg, file_contents=None):
+        self.path_arg = str(path_arg)
+        self.file_contents = file_contents or {}
+
+    def exists(self):
+        print("exists", self.path_arg)
+        return self.path_arg in self.file_contents
+
+    def read_text(self):
+        return self.file_contents.get(self.path_arg, "<wrong, returning arbitrary string>")
+
+    def __str__(self):
+        return self.path_arg
+
+    def __repr__(self):
+        return f"MockPath('{self.path_arg}', file_contents={self.file_contents})"
+
+
+class TestUserToken:
+    @patch.object(ModelRegistry, "get_registered_models")
+    def test_user_token_from_envvar(self, mock_get_registered_models):
+        """Test for user not providing explicitly user_token,
+        reading user token from environment variable."""
+        test_token = "test-token-from-envvar" # noqa: S105
+
+        with tempfile.NamedTemporaryFile(mode="w", delete=False) as token_file:
+            token_file.write(test_token)
+            token_file_path = token_file.name
+
+        try:
+            with patch.dict(os.environ, {"KF_PIPELINES_SA_TOKEN_PATH": token_file_path}):
```

Comment:
nit - I'm thinking we should mostly still with our test frameworks machinery, and use pytest's `monkeypatch.setenv`. I would even use pytest-mock's `mocker` fixture instead of unittest's mocks when possible. But I realize we already have a bit of variation in how we do things. Also fine to keep as is

Reply:
> I'm thinking we should mostly stick with our test framework's machinery, and use pytest's monkeypatch.setenv

I've done this with https://github.com/kubeflow/model-registry/pull/1553/commits/81f87352e0d90d4fc868cbd2dab8bdab3a63573a

> I would even use pytest-mock's mocker fixture instead of directly using unittest's mocks when possible

it doesn't seem to me we import pytest-mock at this time, so we could do that with a follow-up PR to keep this focused on the feat change.